### PR TITLE
Iter3 jason

### PR DESCRIPTION
**Pull request by jayl109**
When adding comments, comments added to all vouchers of that group.
When failing to login, login error message no longer shows after successful login
Also added feature files for these.

### CODE REVIEW

**Comment by WanNJ** [comment]
LGTM

**Comment by jayl109** [comment]
according to fox:
change txn to only record once. 
also try to add some rspec tests for update comments

**Comment by jayl109** [comment]
Rspec tests added for update_comments method of VouchersController

File: app/controllers/vouchers_controller.rb [hunk 1]
```diff
@@ -83,18 +83,22 @@ def create
     if shouldEmail
       email_confirmation(:confirm_reservation, @customer, theshowdate, thenumtoadd)
     end
-    
+
```
Comment (fxdawnn) [inline app/controllers/vouchers_controller.rb:86]: what is this line for?

File: app/controllers/vouchers_controller.rb [hunk 2]
```diff
@@ -83,18 +83,22 @@ def create
...
     redirect_to customer_path(@customer)
   end
 
   def update_comment
-    @voucher.update_attributes(:comments => params[:comments], :processed_by => current_user)
+    vchs = Voucher.find(params[:voucher_ids].split(","))
```
Comment (fxdawnn) [inline app/controllers/vouchers_controller.rb:91]: do you mind make this more readable,like voucher.

File: features/customers/login_error_messages.feature [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  I want to see error messages when login fails
+
+Background:
+
+  Given customer "Tom Foolery" has email "tom@foolery.com" and password "pass"
+  Given I am not logged in
+  When I visit the Login page
+
+Scenario: customer sees incorrect email if they type it in wrong
+  And I fill in "email" with "to@foolery.com"
```
Comment (fxdawnn) [inline features/customers/login_error_messages.feature:14]: There is a easier way to do this. It's something like"As I login as xxx"

**Review by fxdawnn — COMMENTED** [review]
Good, follow TDD.

**Comment by Zhiying12** [comment]
LGTM

File: features/customers/login_error_messages.feature (hunk 3 above)
Comment (jayl109) [reply features/customers/login_error_messages.feature:14]: I'm trying to test what happens if he tries to login with the wrong info, so i cant just do that

File: app/controllers/vouchers_controller.rb (hunk 1 above)
Comment (jayl109) [reply app/controllers/vouchers_controller.rb:86]: just an extra whitespace, dont want to make a commit just to fix it, it doesnt change anything

File: app/controllers/vouchers_controller.rb (hunk 2 above)
Comment (jayl109) [reply app/controllers/vouchers_controller.rb:91]: vchs is pretty readable, just an abbreviated version of vouchers. Fox uses it in his code as well.

**Comment by sanketq** [comment]
lgtm

**Pull request merged**